packages/wrangler/pages/functions/template-plugin.ts

```diff
@@ -132,7 +132,7 @@ export default function (pluginArgs) {
         // https://fetch.spec.whatwg.org/#null-body-status
         return new Response(
           [101, 204, 205, 304].includes(response.status) ? null : response.body,
-          response
+          { ...response, headers: new Headers(response.headers) }
```

Comment:
So this is the fix?

And the other change is just for consistency?

Or is the other change also fixing something?

Reply:
The bug was introduced in `template-worker.ts`. And `template-plugin.ts` was updated for consistency. Both places should now offer mutable headers and keep any duplicate headers.